### DISCUSSION TITLE
pg_dump equivalent - human readable

### DISCUSSION THREAD

**Discussion by DavidPratten** [geldata/gel#2640] [Ideas]
The DDL command 

`DESCRIBE SCHEMA AS SDL;`
dumps the Schema in human readable form.

Is there a `pg_dump` like equivalent which generates human readable text for the non-schema content of the database?

Thanks

David

**Comment by elprans** [comment]
No, there's currently no support for text data dumps.  Do you have a certain use case in mind?

**Reply by DavidPratten** [reply]
Thanks,

The use cases are around 

- learning the DML by example dump, 
- introspection of the database, 
- diffing, 
- source control and,
-  risk mitigation of the brittleness of binary backup forms

David

**Reply by elprans** [reply]
Thanks David.  Some comments below.

> learning the DML by example dump

A hypothetical DML-based dump would necessarily use very primitive `INSERT`-s and `UPDATE`-s, so it's unlikely you'll learn much from those.

> diffing
> source control

It's probably a better idea to use data migrations/fixtures than straight dumps in VCS, since even text dumps aren't guaranteed to be stable (i.e. always generate stuff in diff-friendly ways).  We are working on improving our migration system to make it easier to work with initial/test data loading and updates.

> risk mitigation of the brittleness of binary backup forms

EdgeDB binary dump format is [well-documented](https://www.edgedb.com/docs/internals/protocol/messages#ref-protocol-msg-dump-block), and we go to great lengths to ensure that even dumps from the earliest public releases [stay compatible](https://github.com/edgedb/edgedb/tree/master/tests/dumps/dump01).

> introspection of the database

This is very reasonable.  We have a plan to add a CLI tool for dump file analysis/modification and it'll likely have the ability to perform text output.  This approach lets us keep the dump/restore implementation as simple and efficient as possible, and at the same time provide offline dump manipulation capabilities.
